### PR TITLE
Updated AEC and VAD features

### PR DESCRIPTION
**Pull request by Juniverse**
If VAD is enabled. When player speaking detected, we'll pause the audio playing. So that the AEC quality would be improved.

Also added back the cache list for cancelled items.

INTG-1997, INTG-1980, INTG-1967, INTG-1908

### CODE REVIEW

File: Runtime/Scripts/Interaction/InworldInteraction.cs [hunk 1]
```diff
@@ -37,12 +40,12 @@ public class InworldInteraction : MonoBehaviour
...
-        {
-            get => Random.Range(0, 1);
-            set => m_AnimFactor = value;
-        }
+        public virtual float AnimFactor => Random.Range(0, 1);
 
+        /// <summary>
+        /// Gets this character's audio source
+        /// </summary>
+        public AudioSource PlaybackSource => m_PlaybackSource;
```
Comment (mattkwilson) [inline Runtime/Scripts/Interaction/InworldInteraction.cs:48]: This should probably only exist within InworldAudioInteraction.

File: Runtime/Scripts/Interaction/InworldInteraction.cs
Comment (mattkwilson) [inline Runtime/Scripts/Interaction/InworldInteraction.cs:1]: `m_AnimFactor` is no longer used and can be removed

File: Runtime/Scripts/Interaction/InworldAudioInteraction.cs [hunk 2]
```diff
@@ -4,31 +4,24 @@
  * Use of this source code is governed by the Inworld.ai Software Development Kit License Agreement
  * that can be found in the LICENSE.md file or at https://www.inworld.ai/sdk-license
  *************************************************************************************************/
+
 using Inworld.Packet;
 using UnityEngine;
 using System.Collections;
-
+using System.Linq;
```
Comment (mattkwilson) [inline Runtime/Scripts/Interaction/InworldAudioInteraction.cs:11]: Not necessary

File: Runtime/Scripts/Interaction/InworldAudioInteraction.cs [hunk 3]
```diff
@@ -88,19 +100,23 @@ protected override IEnumerator PlayNextUtterance()
...
-                m_PlaybackSource.PlayOneShot(m_AudioClip);
+                if (audioClip != m_AudioClip)
+                {
+                    m_AudioClip = audioClip;
+                    m_PlaybackSource.clip = m_AudioClip;
+                    m_PlaybackSource.Play();
+                }
                 m_Character.OnInteractionChanged(m_CurrentInteraction.CurrentUtterance.Packets);
-                yield return new WaitUntil(() => !m_PlaybackSource.isPlaying);
+                yield return new WaitUntil(() => m_PlaybackSource.clip == null || m_PlaybackSource.time >= m_PlaybackSource.clip.length - Time.fixedUnscaledDeltaTime);
```
Comment (mattkwilson) [inline Runtime/Scripts/Interaction/InworldAudioInteraction.cs:118]: Why subtract by `Time.fixedUnscaledDeltaTime`?

**Review by mattkwilson — CHANGES_REQUESTED** [review]
Please see comments.

File: Runtime/Scripts/Interaction/InworldAudioInteraction.cs (hunk 3 above)
Comment (Juniverse) [reply Runtime/Scripts/Interaction/InworldAudioInteraction.cs:118]: The audio's playing is in a separate thread other than Unity's game lifecycle. Which means, there is a highly chance that in the last frame, the `m_PlaybackSource.time` is smaller than  `m_PlaybackSource.clip.length`, (the difference is less than deltaTime), but in the next frame, it stopped playing and return 0.

**Pull request closed without merging**